server/routes/users.js

```diff
@@ -2,14 +2,272 @@
 var express = require('express');
 var router = express.Router();
 
+
+// Require database adapter file (not node-postgres directly)
+const db = require('../db')
+
 // Returns test string to verify that Users server is running.
 router.get('/ping', function(req, res, next) {
   res.send('pong - Users API');
 });
 
 // GET users listing.
 router.get('/', function(req, res, next) {
-    // TODO
+    db.query('SELECT id FROM swe_user', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+
+//having an issue with concat
+/*
+//GET all user names.
+router.get('/', function(req, res, next) {
+    db.query('SELECT CONCAT(first_name, ' ', last_name) FROM swe_user', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+*/
+
+
+//GET all user emails.
+router.get('/emails', function(req, res, next) {
+
+    db.query('SELECT email FROM swe_user', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+ 
+//Get all user university IDs.
+router.get('/ids', function(req, res, next) {
+
+    db.query('SELECT university_id FROM swe_user;', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+
+//Get user info by user_id
+router.get('/<user_id>', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT * FROM swe_user WHERE id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get whether user is admin or not
+router.get('/:user_id/admin', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT is_admin FROM swe_user WHERE id = $1', [id], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+//Get a user's past events-HELP NEEDED
+router.get('/:user_id/past', function(req, res, next) {
+  const id = req.params.user_id;
+  db.query('SELECT event_id FROM event_checkin INNER JOIN event ON event_checkin.event_id = event.fb_id WHERE event_checkin.user_id = 3 AND event.ends_at < now() UNION SELECT event_id FROM event_registration INNER JOIN event ON event_registration.event_id = event.fb_id WHERE event_registration.user_id = 3 AND event.ends_at < now()', [], (err, result) => {
```

Comment:
Reformat this to be readable, test the route to make sure query works as expected, and use the parameter value.